app/client/src/components/joinButton.js

```diff
@@ -17,39 +17,71 @@
 import Button from "@mui/material/Button";
 import React, { useEffect, useState } from "react";
 import { connect } from "react-redux";
-import { PropTypes } from "prop-types"
-import { joinCommunity } from "../redux/ducks/communityDuck"
+import { PropTypes } from "prop-types";
+import { joinCommunity, leaveCommunity } from "../redux/ducks/communityDuck";
 
 const JoinButton = (props) => {
-  const [flag, setFlag] = useState(false);
+  const { community, communities } = props;
 
+  const isUserInCommunity = (communityName) => {
+    return (
+      communities.usersCommunities.find(
+        (element) => element.title === communityName
+      ) !== undefined
+    );
+  };
+
+  const [isInCommunity, setIsInCommunity] = useState(
+    isUserInCommunity(community.title)
+  );
 
-  const handleClick = () => {
-    setFlag(!flag);
-    props.joinCommunity(props.name)
+  const handleJoin = async () => {
```

Comment:
Disregard my previous comment. I see that you have implemented the handleJoin here with a conditional to check for the success of the joinCommunity function.